src/py_pecg/filtering_module.py

```diff
@@ -0,0 +1,564 @@
+
+    elif fs == 200:
+        # Interpolate by 4
+        interp_params = [
+            False,
+            (4, 3, 0.4),
+            (4, 7, 0.4),
+            (4, 7, 0.4),
+            (4, 7, 0.4),
+        ]
+        quadratic_filter_array = apply_interpolation(
+            quadratic_filter_array, interp_params
+        )
+        for i in range(0, len(quadratic_filter_array)):
+            quadratic_filter_array[i] = quadratic_filter_array[i][1:-4]
+
+        # Downsample by 5
+        quadratic_filter_array[0] = 1.1 * np.array([5 / 4, -5 / 4])
+        for i, ind in enumerate([False, 3, 4, 1, 1]):
+            if i != 0:
+                quadratic_filter_array[i] = quadratic_filter_array[i][ind::5]
+
+    elif fs == 360:
+        # Interpolate by 6
+        interp_params = [(6, 1, 0), (6, 3, 0), (6, 7, 0), (6, 7, 0), (6, 7, 0)]
+        quadratic_filter_array = apply_interpolation(
+            quadratic_filter_array, interp_params
+        )
+        for i in range(0, len(quadratic_filter_array)):
+            quadratic_filter_array[i] = quadratic_filter_array[i][1:-6]
+
+        # Downsample by 5
+        for i, ind in enumerate([3, 0, 4, 2, 2]):
+            quadratic_filter_array[i] = quadratic_filter_array[i][ind::5]
+
+        # Interpolate by 6 again
+        interp_params = [(6, 1, 0), (6, 3, 0), (6, 7, 0), (6, 7, 0), (6, 7, 0)]
+        quadratic_filter_array = apply_interpolation(
+            quadratic_filter_array, interp_params
+        )
+        for i in range(0, len(quadratic_filter_array)):
+            quadratic_filter_array[i] = quadratic_filter_array[i][1:-6]
+
+        # Final Downsample by 5
+        for i, ind in enumerate([1, 4, 3, 3, 3]):
+            quadratic_filter_array[i] = quadratic_filter_array[i][ind::5]
+
+    return quadratic_filter_array, messages
+
+
+def wavelet_transform(
+    signal: np.ndarray,
+    quadratic_filter_array: np.ndarray,
+) -> np.ndarray:
+    """
+    Calculates the wavelet transform of a signal using quadratic spline wavelet.
+    It calculates wavelets in scales of 1 to 4.
+
+    Args:
+        signal (np.ndarray): The target signal designated for filtering.
+        quadratic_filter_array (np.ndarray): Qquadratic spline filter bank.
+
+    Returns:
+        wavelet_transform_matrix (np.ndarray): Resultant Wavelet Transform Matrix.
+    """
+    if signal.size == 0:
+        raise ValueError("Input is an empty array.")
+    if not isinstance(quadratic_filter_array, list) or any(
+        arr.size == 0 for arr in quadratic_filter_array
+    ):
+        raise ValueError(
+            "quadratic_filter_array should be a list of 1D numpy arrays."
+        )
+    if signal.squeeze().ndim > 1 or any(
+        not isinstance(arr, np.ndarray) or arr.ndim != 1
+        for arr in quadratic_filter_array
+    ):
+        raise ValueError("Input is a higher-dimensional array, not a matrix.")
+
+    wavelet_transform_matrix = np.zeros((len(signal), 5))
```

Comment:
The `5` hardcoding might be confusing for other devs, let's create a constant for it